sha1-name.c

```diff
@@ -1582,31 +1606,52 @@ static void diagnose_invalid_index_path(int stage,
 	if (pos < active_nr) {
 		ce = active_cache[pos];
 		if (ce_namelen(ce) == fullname.len &&
-		    !memcmp(ce->name, fullname.buf, fullname.len))
+		    !memcmp(ce->name, fullname.buf, fullname.len)) {
+			if (gently)
+				return -1;
 			die("Path '%s' is in the index, but not '%s'.\n"
 			    "Did you mean ':%d:%s' aka ':%d:./%s'?",
 			    fullname.buf, filename,
 			    ce_stage(ce), fullname.buf,
 			    ce_stage(ce), filename);
```

Comment:
And also here, I could imagine that the error message might need to be printed even when returning -1.

OTOH this might merit turning `gently` into `flags` so that it can be gentle and/or quiet. Dunno. What do you think, @ungps?

Reply:
Oh! I just saw that there is already a `GET_OID_QUIETLY`! So maybe we want to accept a parameter `flags` here, and also fix the issue where the `die()` prints something despite `flags & GET_OID_QUIETLY` possibly being non-zero.

I think I would do this, as a separate patch that I would insert before this one: add the `int flags` parameter and inserting `if (flags & GET_OID_QUIETLY) exit(128);` before the `die()` call.